### ISSUE TITLE
Uncaught TypeError: Rollbar\Payload\Frame::__construct(): Argument #1 ($filename) must be of type string

### ISSUE THREAD

**Issue by thomasme** [rollbar/rollbar-php#548]
When logging a message (any level) with explicit Rollbar::log call, this error happens.

On PHP 8.0.12 with both 3.0.0 and master@ad22e3589116ff5958c3b6ff35c65d959266414d and Rollbar defaults.

Does not seem to happen when issues are caught through default error handlers.

Does not happen when `capture_error_stacktraces` is set to `false` on init.

```
<b>Fatal error</b>:  Uncaught TypeError: Rollbar\Payload\Frame::__construct(): Argument #1 ($filename) must be of type string, null given, called in /srv/vendor/rollbar/rollbar/src/DataBuilder.php on line 479 and defined in /srv/vendor/rollbar/rollbar/src/Payload/Frame.php:18
Stack trace:
#0 /srv/vendor/rollbar/rollbar/src/DataBuilder.php(479): Rollbar\Payload\Frame-&gt;__construct(NULL)
#1 /srv/vendor/rollbar/rollbar/src/DataBuilder.php(454): Rollbar\DataBuilder-&gt;makeFrames(Object(TypeError), false)
#2 /srv/vendor/rollbar/rollbar/src/DataBuilder.php(425): Rollbar\DataBuilder-&gt;makeTrace(Object(TypeError), false)
#3 /srv/vendor/rollbar/rollbar/src/DataBuilder.php(406): Rollbar\DataBuilder-&gt;getExceptionTrace(Object(TypeError))
#4 /srv/vendor/rollbar/rollbar/src/DataBuilder.php(375): Rollbar\DataBuilder-&gt;getBody(Object(TypeError), Array)
#5 /srv/vendor/rollbar/rollbar/src/Config.php(732): Rollbar\DataBuilder-&gt;makeData('...', Object(TypeError), Array)
#6 /srv/vendor/rollbar/rollbar/src/RollbarLogger.php(215): Rollbar\Config-&gt;getRollbarData('...', Object(TypeError), Array)
#7 /srv/vendor/rollbar/rollbar/src/RollbarLogger.php(120): Rollbar\RollbarLogger-&gt;getPayload('...', '...', Object(TypeError), Array)
#8 /srv/vendor/rollbar/rollbar/src/Handlers/ExceptionHandler.php(28): Rollbar\RollbarLogger-&gt;log('...', Object(TypeError), Array)
#9 [internal function]: Rollbar\Handlers\ExceptionHandler-&gt;handle(Object(TypeError))
#10 {main}
  thrown in <b>/srv/vendor/rollbar/rollbar/src/Payload/Frame.php</b> on line <b>18</b><br />
```

**Comment by bishopb** [comment]
Thanks for the detailed report, @thomasme! I can confirm your analysis:

[`Rollbar\DataBuilder::makeFrames()` may set a `$filename` to `null`][1] and later pass that to the `Rollbar\Payload\Frame` constructor:
```
$filename = isset($frameInfo['file']) ? $frameInfo['file'] : null;
// ...
$frame = new Frame($filename);
```

[The `Frame`, however, does not allow for `null` filename.][2]

```
public function __construct(private string $filename)
```

[1]:https://github.com/rollbar/rollbar-php/blob/master/src/DataBuilder.php#L471
[2]:https://github.com/rollbar/rollbar-php/blob/master/src/Payload/Frame.php#L18

Planning a fix for the upcoming version 3.0.1.

**Comment by Honzy** [comment]
Hello, is there a way how to bypass this error?

**Comment by victoriafrench** [comment]
same question here, it is taking is down in production

**Comment by Honzy** [comment]
I downgraded one version down and it's working fine.

**Comment by shin1x1** [comment]
same problem here. I changed version to `dev-master` (https://github.com/rollbar/rollbar-php/commit/bdcb1d7922c0ae4a1b09bf93f8e03fb73b8f35a7) and it's working fine on PHP 8.1.

**Comment by danielmorell** [comment]
This was fixed in v3.1.0.
